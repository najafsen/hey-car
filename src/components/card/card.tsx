import styled from "styled-components";
import { PropsWithChildren } from "../../types/global";

const CardHeader = styled.h4`
  color: #777777;
  margin-bottom: 12px;
`;

const CardContent = styled.section`
  color: #c2c2c2;
`;

interface CardProps extends PropsWithChildren {
  header: string;
}

export const Card = ({ header, children }: CardProps) => (
  <article>
    <CardHeader>{header}</CardHeader>
    <CardContent>{children}</CardContent>
  </article>
);
