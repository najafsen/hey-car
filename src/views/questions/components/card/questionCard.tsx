import { Link as RouterLink } from "react-router-dom";
import styled from "styled-components";
import { Card } from "../../../../components/card/card";
import { questionsPageBasePath } from "../../questions.router";

const Line = styled.div`
  margin: 4px 0;
`;

const Link = styled(RouterLink)`
  text-decoration: none;
  padding: 4px;

  :hover {
    background-color: #fafafa;
  }
`;

interface QuestionCardProps {
  id: string;
  title: string;
  published: string;
  choicesCount: number;
}

export const QuestionCard = ({
  id,
  title,
  published,
  choicesCount,
}: QuestionCardProps) => (
  <Link to={`${questionsPageBasePath}/${id}`}>
    <Card header={title}>
      <Line>{published}</Line>
      <Line>
        Choices count: <strong>{choicesCount}</strong>
      </Line>
    </Card>
  </Link>
);
