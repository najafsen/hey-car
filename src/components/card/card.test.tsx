import { render } from "@testing-library/react";
import { Card } from "./card";

describe("Card", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Card header="test">Content</Card>);
    expect(asFragment()).toMatchSnapshot();
  });
});
