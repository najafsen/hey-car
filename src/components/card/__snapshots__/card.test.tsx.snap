// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<DocumentFragment>
  <article>
    <h4
      class="sc-bdnxRM IvBzo"
    >
      test
    </h4>
    <section
      class="sc-gtsrHT gmVwmo"
    >
      Content
    </section>
  </article>
</DocumentFragment>
`;
